### PR TITLE
refactor: TableArgs preserve info of positioned and named args

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1.  refactor as refactor: TableArgs preserve info of positioned and named args
2.  add some common functions to help parse/check
3.  small fixes.

Closes https://github.com/datafuselabs/databend/issues/9914

### CODE REVIEW

File: src/query/catalog/src/table_args.rs [hunk 1]
```diff
@@ -12,6 +12,62 @@
...
+    }
+
+    pub fn new_positioned(args: Vec<Scalar>) -> Self {
+        Self {
+            positioned: args,
+            named: HashMap::new(),
+        }
+    }
+
+    pub fn expect_all_positioned(
```
Comment (BohuTANG) [inline src/query/catalog/src/table_args.rs:39]: Add more comments for this function? From the name I can not get what its usage is.

Comment (BohuTANG) [reply src/query/catalog/src/table_args.rs:39]: Others LGTM 👍

Comment (youngsofun) [reply src/query/catalog/src/table_args.rs:43]: done

File: src/query/sql/src/planner/binder/table_args.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        .map(|scalar| match scalar {
+            ScalarExpr::ConstantExpr(ConstantExpr { value, .. }) => Ok(check_literal(&value).0),
+            _ => Err(ErrorCode::Unimplemented(format!(
+                "Unsupported table argument type: {:?}",
+                scalar
+            ))),
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+    let named_args: Vec<(String, Scalar)> = named_args
```
Comment (leiysky) [inline src/query/sql/src/planner/binder/table_args.rs:52]: Can we just build a `HashMap<String, Scalar>` here?

Comment (youngsofun) [reply src/query/sql/src/planner/binder/table_args.rs]: done, thank you!

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**